src/irmin-pack/IO.ml

```diff
@@ -127,7 +142,7 @@ module Unix : S = struct
 
   let name t = t.file
 
-  let header = 16L (* offset + version *)
+  let header = 24L (* offset + version + generation *)
```

Comment:
Does this change the on-disk format?

Reply:
Yes, it does. (I was going to write a comment once I repaired [PR 168 on index](https://github.com/mirage/index/pull/168) as the two are related).
 
Clearing the store messes up the [sync](https://github.com/mirage/irmin/pull/940/files#diff-1beb7e2142825f197755316a39d00699R170) of RO instances. There are two solutions: either we always refill cache buffers of RO, which is what I do in [the layer store](https://github.com/mirage/irmin/pull/940/files#diff-1beb7e2142825f197755316a39d00699R170) to be able to test it. But this is slow. 

The second solution is what I'm proposing here: add a generation field, as in index, to keep track of when the file was rewritten (by `merge` or `clear` in index). I'm still testing if we can use generation to keep track of clear in index, but I think that would be a nice solution. And we eventually do want to use the same IO for both pack and index. But as you point it out, it needs a format change.